source/common/secret/sds_api.cc

```diff
@@ -154,14 +168,17 @@ void SdsApi::onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason reaso
   init_target_.ready();
 }
 
-absl::Status SdsApi::validateUpdateSize(int num_resources) {
-  if (num_resources == 0) {
+absl::Status SdsApi::validateUpdateSize(uint32_t added_resources_num,
+                                        uint32_t removed_resources_num) {
+  if (added_resources_num + removed_resources_num == 0) {
```

Comment:
nit: prefer `(A == 0 && B == 0)`, to avoid overflow (not that it should ever happen in reality, but just to be on the safe side).